pypeman/channels.py

```diff
@@ -455,6 +457,28 @@ def graph_dot(self, end=''):
         for end, sub in after:
             sub.graph_dot(end=end)
 
+    def jsonable_msg_info_for_admin(self, msg):
```

Comment:
Why this is not a `Message` method ?

Reply:
Each channel handles different kinds of messages.
I imagine the usecase where a channel wants to have a different representation of a message.

Let's say an MLLP channel would prefer to have perhaps a partially preparsed representation for the web interface,
A file watcher channel might not display the contents of the file, but more stats about the file.

An alternative would of course be to put this method into the channel and let certain endpoints / channels subclass the Message object. (Though this might imply more data copies)

Reply:
Seems fair. 

New proposal : you add the method to the `Message` class but you allow to redefine default payload_encoder with an argument. Then you add a method to the channel that call the `Message` method with the appropriate encoder. Later (i.e. in another PR) we can add a way to custom encoder by a channel argument.

What do you think ?

I think you can use current `.to_dict()` method and add the correct argument to achieve that ?

Reply:
should the param for message_payload_encoder be 
a class, an instance (which must have the method encode or just a function that encodes).

Will make a first start with an instance, but I am open.

Reply:
pls check pypeman/message.py and pypeman/helpers/serializers.py ( 297b219d5c9b04d617ddf1c9e758fb52748a4aaa )

Is this what you mean?

Reply:
I don't find this commit, do you have any link ? (github seems not doing proper link for commit ids)

Reply:
My bad. pushed only to local gitlab server and not to github

Just check https://github.com/klausfmh/pypeman/commit/297b219d5c9b04d617ddf1c9e758fb52748a4aaa

Reply:
In fact best to recheck all the deltas of this MR

Reply:
reminder to continue review :-)
I think I handled all the FB. pls correct me if wrong

Reply:
Yes that's what i think for message but part  :

> - Add a message_payload_encoder property to channel with b64 default encoder with default value to b64encoder,
> - Use this encoder in the .list_message() while calling .to_json() to get your formatted message.

is not yet done ?

Reply:
I think `jsonable_msg_info_for_admin` channel method should not exists anymore at the end.